canister/src/validation.rs

```diff
@@ -76,12 +76,21 @@ mod test {
         #![proptest_config(ProptestConfig::with_cases(10))]
         #[test]
         fn validation_context(
-            stability_threshold in 1..150u32,
+            stability_threshold_start in 1..150u32,
             num_blocks in 1..250u32,
         ) {
             let num_transactions_in_block = 1;
             let network = Network::Regtest;
-            let blocks = build_chain(network, num_blocks, num_transactions_in_block);
+
+            let stability_threshold = stability_threshold_start + 1;
```

Comment:
Why do we need to increment the stability threshold by one?

Reply:
If we have the tree with only the root with difficulty 1, `fn depth()` will return 0, while `fn difficulty_based_depth()` will return 1. The same follows in all other cases. Hence to make the same behavior we need to increment the stability threshold by one.

Reply:
The specification says that depth is a function of the successors, so `difficulty_based_depth()` should also return 0 because there are no successors, right?

Reply:
Both `depth()` and `difficulty_based_depth()` are called on the children of the root. If the root does not have children then they will never be called. But in case the child of the root has difficulty 1, and does not have any successors `depth()` will return 0, while `difficulty_based_depth()` will return 1.

Reply:
I don't understand why it returns 1. If this child has no successors, the sum over the difficulties must be 0. If we normalize, we get 0/1 = 0.

Reply:
But `child`'s difficulty is 1, hence the path with the max difficulty in the subtree that has `child` as the root has difficulty 1.

Reply:
(unresolving) I'm still not clear on why we're touching this test. I don't see what test case fails by not incrementing the stability threshold by one, nor is it necessary anymore to mock the difficulty of the blocks below.

Reply:
You are right, we do not need it now, it is changed.